File: LeetCode/Array/JS/Leetcode540-有序数组中的单一元素.js
var singleNonDuplicate = function(nums) {
    nums.push(nums[-1] + 1);
    let l = 0,r = Math.floor(nums.length / 2 - 1);
    while(l < r)
    {
        let mid = (l + r) >> 1;
        if(nums[mid * 2] != nums[mid * 2 + 1]) r = mid;
        else l = mid + 1;
    }
    return nums[l * 2];
};

File: LeetCode/Array/Golang/Leetcode540-有序数组中的单一元素.go
package sbgo

func singleNonDuplicate(nums []int) int {
	nums = append(nums, nums[len(nums)-1]+1)
	l, r := 0, len(nums)/2-1
	for l < r {
		mid := (l + r) >> 1
		if nums[mid*2] != nums[mid*2+1] {
			r = mid
		} else {
			l = mid + 1
		}
	}
	return nums[l*2]
}
